packages/core/src/utils/LockManager.ts

```diff
@@ -0,0 +1,47 @@
+import { v4 as uuidv4 } from 'uuid';
+
+class LockManager {
```

Comment:
Adding comment/overview/description of this class would be nice, to help retain the context

Reply:
Added